### PR TITLE
support HF models/dataset mapping

### PR DESCRIPTION
**Pull request by ganisback**
Introduced a new query parameter 'mirror', available value
auto: will check git resource from both hf and csghub mirror, default value
hf: only check the hf mirror to find the correct repo id
csghub: only check the csghub mirror to find the correct repo id

### CODE REVIEW

File: builder/store/database/mirror.go [hunk 1]
```diff
@@ -80,6 +80,35 @@ func (s *MirrorStore) FindByRepoPath(ctx context.Context, repoType types.Reposit
...
 
+func (s *MirrorStore) FindWithMapping(ctx context.Context, repoType types.RepositoryType, namespace, name string, mapping types.Mapping) (*Mirror, error) {
+	var mirror Mirror
+	var err error
+	if mapping == types.CSGHubMapping {
+		return s.FindByRepoPath(ctx, repoType, namespace, name)
+	} else if mapping == types.HFMapping {
+		err = s.db.Operator.Core.NewSelect().
+			Model(&mirror).
+			Relation("Repository").
```
Comment (pulltheflower) [inline builder/store/database/mirror.go:91]: The `Relation` method is already called, which automatically joins `repositories` table, so the `Join` method is not needed

File: builder/store/database/mirror.go [hunk 2]
```diff
@@ -80,6 +80,35 @@ func (s *MirrorStore) FindByRepoPath(ctx context.Context, repoType types.Reposit
...
+			Join("JOIN repositories AS r ON mirror.repository_id = r.id ").
+			Where("mirror.source_repo_path=?", fmt.Sprintf("%s/%s", namespace, name)).
+			Where("r.repository_type=?", repoType).
+			Scan(ctx)
+	} else {
+		// auto mapping
+		err = s.db.Operator.Core.NewSelect().
+			Model(&mirror).
+			Relation("Repository").
```
Comment (pulltheflower) [inline builder/store/database/mirror.go:100]: Same as above.

File: builder/store/database/mirror.go (hunk 1 above)
Comment (ganisback) [reply builder/store/database/mirror.go:91]: Fixed

File: builder/store/database/mirror.go (hunk 2 above)
Comment (ganisback) [reply builder/store/database/mirror.go:99]: fixed

**Review by pulltheflower — APPROVED** [review]

**Pull request merged**